From 5b0565a72048f1b8f516bdc1cd3a675e92bdc61f Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Wed, 11 Jul 2012 11:01:33 +0530
Subject: [PATCH] arm/dts: DT support for EHRPWM and ECAP device.

Adds DT support for EHRPWM and ECAP APWM device. Also sets size of
pwm-cells to 3 to support -  <PWM instance number>, <PWM period> in
nano-seconds and <PWM polarity>.

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 arch/arm/boot/dts/am33xx.dtsi |   60 +++++++++++++++++++++++++++++++++++++++++
 1 files changed, 60 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/am33xx.dtsi b/arch/arm/boot/dts/am33xx.dtsi
index 0b0cf02..cb0b83d 100644
--- a/arch/arm/boot/dts/am33xx.dtsi
+++ b/arch/arm/boot/dts/am33xx.dtsi
@@ -412,5 +412,65 @@
 			interrupts = <20 21 22 23 24 25 26 27>;
 			status = "disabled";
 		};
+
+		ehrpwm0: ehrpwm@48300200 {
+			compatible	= "ti,omap2-ehrpwm";
+			reg = <0x48300200 0x100 0x48300000 0x10>;
+			interrupt-parent = <&intc>;
+			interrupt = <86 58>;
+			ti,hwmods = "ehrpwm0";
+			#pwm-cells = <3>;
+			status = "disabled";
+		};
+
+		ehrpwm1: ehrpwm@48302200 {
+			compatible	= "ti,omap2-ehrpwm";
+			reg = <0x48302200 0x100 0x48302000 0x10>;
+			interrupt-parent = <&intc>;
+			interrupt = <87 59>;
+			ti,hwmods = "ehrpwm1";
+			#pwm-cells = <3>;
+			status = "disabled";
+		};
+
+		ehrpwm2: ehrpwm@48304200 {
+			compatible	= "ti,omap2-ehrpwm";
+			reg = <0x48304200 0x100 0x48304000 0x10>;
+			interrupt-parent = <&intc>;
+			interrupt = <39 60>;
+			ti,hwmods = "ehrpwm2";
+			#pwm-cells = <3>;
+			status = "disabled";
+		};
+
+		ecap0: ecap@48300100 {
+			compatible	= "ti,omap2-ecap";
+			reg = <0x48300100 0x80 0x48300000 0x10>;
+			interrupt-parent = <&intc>;
+			interrupt = <31>;
+			ti,hwmods = "ecap0";
+			#pwm-cells = <3>;
+			status = "disabled";
+		};
+
+		ecap1: ecap@48302100 {
+			compatible	= "ti,omap2-ecap";
+			reg = <0x48302100 0x80 0x48302000 0x10>;
+			interrupt-parent = <&intc>;
+			interrupt = <47>;
+			ti,hwmods = "ecap1";
+			#pwm-cells = <3>;
+			status = "disabled";
+		};
+
+		ecap2: ecap@48304100 {
+			compatible	= "ti,omap2-ecap";
+			reg = <0x48304100 0x80 0x48304000 0x10>;
+			interrupt-parent = <&intc>;
+			interrupt = <61>;
+			ti,hwmods = "ecap2";
+			#pwm-cells = <3>;
+			status = "disabled";
+		};
 	};
 };
-- 
1.7.7.6

